test/benchmarks/mongoBench/runner.js

```diff
@@ -180,12 +165,22 @@ class Runner {
     let time = performance.now() - start;
     let count = 1;
 
-    while (time < maxExecutionTime && (time < minExecutionTime || count < minExecutionCount)) {
-      await benchmark.beforeTask.call(ctx);
-      const executionTime = await timeTask(benchmark.task, ctx);
-      rawData.push(executionTime);
-      count++;
-      time = performance.now();
+    if (benchmark._taskType === 'sync') {
+      while (time < maxExecutionTime && (time < minExecutionTime || count < minExecutionCount)) {
```

Comment:
is there a downside to doing the if/else earlier, something like `const timeTask = becnhmark._taskType === 'sync' ? timeSyncTask : timeAsyncTask` and keeping the while loop the same since `await` works equally well in front of sync and async things?

Reply:
Nope. that works too

Reply:
Done!